clients/tabby-agent/src/codeCompletion/index.ts

```diff
@@ -838,6 +840,38 @@ export class CompletionProvider implements Feature {
     return snippets;
   }
 
+  //get all recently opened files from the file tracker
+  private async collectSnippetsFromOpenedFiles(): Promise<
+    { filepath: string; offset: number; text: string; score: number }[] | undefined
+  > {
+    const config = this.configurations.getMergedConfig();
+    if (!config.completion.prompt.collectSnippetsFromRecentOpenedFiles.enabled) {
+      return undefined;
+    }
+    this.logger.debug("Starting collecting snippets from opened files.");
+    const recentlyOpenedFiles = this.fileTracker.getAllFilesWithoutActive();
+    const codeSnippets: { filepath: string; offset: number; text: string; score: number }[] = [];
+    recentlyOpenedFiles.forEach((file) => {
+      const doc = this.documents.get(file.uri);
+      if (doc) {
+        file.lastVisibleRange.forEach((range) => {
+          const startOffset = doc.offsetAt(range.start);
+          const text = doc.getText(range);
```

Comment:
Use all text in the visible range may create a very long code snippet.  
I suggest to consider using `CodeSearchEngine` to split visible range into chunks when updating, and make a text search query when reading.

Reply:
How about:
1. find the line cursor focused on  (or visible middleline).
2. collect content around line from #1, and go as much as it could with a length limit?

Reply:
now it's last visible range's middle line with configurable block size